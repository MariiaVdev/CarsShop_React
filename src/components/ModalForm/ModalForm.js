import styles from './ModalForm.module.scss';
import { useDispatch, useSelector } from 'react-redux';
import { setIsOpenFormAC } from '../../store/form/actionCreators';
import OrderedGoods from '../OrderedGoods/OrderedGoods';
import OrderForm from '../OrderForm/OrderForm';

const ModalForm = () => {
    const isOpenForm = useSelector(store => store.form.isOpenForm);
    const dispatch = useDispatch();

    if (!isOpenForm) {
        return null;
    }
    const closeForm = () => { dispatch(setIsOpenFormAC(false)) };
    return (
        <div className={styles.form}>
            <div data-testid='ModalForm-bgd' className={styles.formBackground} onClick={closeForm}></div>
            <div className={styles.formMainContainer}>
                <div className={styles.formHeader}>
                    <h3>Ordering</h3>
                    <button data-testid='ModalForm-btn' className={styles.formClose} onClick={closeForm}></button>
                </div>
                <div className={styles.formContentWrapper}>
                    <OrderedGoods />
                    <OrderForm />
                </div>

            </div>
        </div>

    )
}


export default ModalForm;