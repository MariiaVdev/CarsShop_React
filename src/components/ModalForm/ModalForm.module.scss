.form {
    position: fixed;
    top: 0;
    left: 0;
    width: 100vw;
    height: 100vh;
    transform: translate(0, 0);
    z-index: 1000;
    display: flex;
    justify-content: center;
    align-items: center;
    font-family: sans-serif;
}

.formBackground {
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    left: 0;
    background: rgba(0, 0, 0, .8);
    cursor: pointer;
    z-index: 1;
}

.formMainContainer {
    z-index: 2;
    background: #bed1be;
    border: none;
    border-radius: 5px;
    box-sizing: border-box;
    padding-bottom: 20px;
    position: relative;
    display: flex;
    flex-direction: column;
    width: 600px;
    height: 590px;
}


.formHeader {
    background-color: #276627;
    height: 50px;

    h3 {
        color: white;
        font-size: 28px;
        float: left;
        margin: 12px 0px 0px 20px;
        text-transform: uppercase;
    }

    .formClose {
        position: absolute;
        top: 15px;
        right: 30px;
        width: 28px;
        height: 28px;
        background-color: transparent;
        border: none;
        background-repeat: no-repeat;
        background-position: center center;
        background-image: url("../../../public/img/close.png");
        float: right;

        &:hover {
            box-shadow: 0 0 4px 0 rgba(34, 60, 80, 0.6);
            filter: brightness(1);
        }

        &:active {
            box-shadow: 0 0 4px 0 rgba(34, 60, 80, 0);
        }
    }
}

.formContentWrapper {
    margin-bottom: 30px;
    box-sizing: border-box;
    padding: 6px 10px;
    resize: none;
    height: 100px;
    border-radius: 4px;
    text-align: center;

    p {
        font-size: 18px;
        color: white;
    }

}