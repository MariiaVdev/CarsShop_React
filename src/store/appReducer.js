import {combineReducers} from "redux";
import modalReducer from "./modal/modalReducer";
import carsReducer from './cars/carsReducer';
import cardReducer from './card/cardReducer';
import cartReducer from './cart/cartReducer';
import formReducer from "./form/formReducer";





const appReducer = combineReducers({
    modal: modalReducer,
    cars: carsReducer,
    card: cardReducer,
    cart: cartReducer,
    form: formReducer
})


export default appReducer;