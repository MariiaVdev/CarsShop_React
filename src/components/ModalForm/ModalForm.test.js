import ModalForm from './ModalForm';
import { render, screen, fireEvent } from '@testing-library/react';
import renderWithProviders from '../../utils/utilsModalForm/utilsModalFormTest';
import { Provider } from 'react-redux';
import store from '../../store/index'

const Component = () => {
    return (
        <Provider store={store}>
            <ModalForm />
        </Provider>
    )

}
jest.mock('../OrderedGoods/OrderedGoods', () => {
    return {
    __esModule: true,
    default: () => {
      return <span>OrderedGoods</span>;
    },
  };})
  jest.mock('../OrderForm/OrderForm', () => {
    return {
    __esModule: true,
    default: () => {
      return <span>OrderForm</span>;
    },
  };})
describe('Modal renders', () => {
    test('should Modal no render without isOpen state', () => {
        const { asFragment } = render(<Component />);
        expect(asFragment()).toMatchSnapshot();
    })
    test('should Modal render when state isOpen set true', () => {
        const { asFragment } = renderWithProviders(<ModalForm/>);
        expect(asFragment()).toMatchSnapshot();

    })
})
describe('Modal close', () => {
    test('should Modal background close work', () => {
        renderWithProviders(<ModalForm/>);
        const bgd = screen.getByTestId("ModalForm-bgd");
        expect(screen.getByTestId("ModalForm-bgd")).toBeInTheDocument();
        fireEvent.click(bgd)
        expect(screen.queryByTestId("ModalForm-bgd")).not.toBeInTheDocument();
    })
    test('should Modal header button close work', () => {
        renderWithProviders(<ModalForm/>);
        const btn = screen.getByTestId("ModalForm-btn");
        expect(screen.getByTestId("ModalForm-btn")).toBeInTheDocument();
        fireEvent.click(btn)
        expect(screen.queryByTestId("ModalForm-btn")).not.toBeInTheDocument();
    })

})
