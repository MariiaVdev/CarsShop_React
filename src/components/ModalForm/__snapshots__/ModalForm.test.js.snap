// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders should Modal no render without isOpen state 1`] = `<DocumentFragment />`;

exports[`Modal renders should Modal render when state isOpen set true 1`] = `
<DocumentFragment>
  <div
    class="form"
  >
    <div
      class="formBackground"
      data-testid="ModalForm-bgd"
    />
    <div
      class="formMainContainer"
    >
      <div
        class="formHeader"
      >
        <h3>
          Ordering
        </h3>
        <button
          class="formClose"
          data-testid="ModalForm-btn"
        />
      </div>
      <div
        class="formContentWrapper"
      >
        <span>
          OrderedGoods
        </span>
        <span>
          OrderForm
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
